programs/solana/programs/ics26-router/src/instructions/client.rs

```diff
@@ -123,12 +138,33 @@ pub fn update_client(ctx: Context<UpdateClient>, _client_id: String, active: boo
         RouterError::UnauthorizedSender
     );
 
-    client.active = active;
+    require!(
+        params.client_program_id.is_some()
+            || params.counterparty_info.is_some()
+            || params.active.is_some(),
+        RouterError::InvalidMigrationParams
+    );
 
-    emit!(ClientStatusUpdatedEvent {
-        client_id: client.client_id.clone(),
-        active,
-    });
+    if let Some(new_program_id) = params.client_program_id {
+        client.client_program_id = new_program_id;
+    }
+
+    if let Some(new_counterparty_info) = params.counterparty_info.clone() {
+        require!(
+            !new_counterparty_info.merkle_prefix.is_empty(),
+            RouterError::InvalidCounterpartyInfo
+        );
+        client.counterparty_info = new_counterparty_info;
+    }
+
+    if let Some(new_active) = params.active {
+        client.active = new_active;
+
+        emit!(ClientStatusUpdatedEvent {
+            client_id: client.client_id.clone(),
+            active: new_active,
+        });
```

Comment:
You should just emit the entire `Client` at the end of execution instead of this event in the if statement